pkg/yurtmanager/controller/yurtappoverrider/yurtappoverrider_controller.go

```diff
@@ -0,0 +1,223 @@
+		Reconciler: r, MaxConcurrentReconciles: concurrentReconciles,
+	})
+	if err != nil {
+		return err
+	}
+
+	yurtappoverriderPredicates := predicate.Funcs{
+		CreateFunc: func(evt event.CreateEvent) bool {
+			obj, ok := evt.Object.(*appsv1alpha1.YurtAppOverrider)
+			if !ok {
+				return ok
+			}
+			if err := r.(*ReconcileYurtAppOverrider).updatePools(obj); err != nil {
+				klog.Errorf("fail to update deployments belonging to obj: %v", err)
+			}
+			return true
+		},
+		DeleteFunc: func(evt event.DeleteEvent) bool {
+			obj, ok := evt.Object.(*appsv1alpha1.YurtAppOverrider)
+			if !ok {
+				return ok
+			}
+			if err := r.(*ReconcileYurtAppOverrider).updatePools(obj); err != nil {
+				klog.Errorf("fail to update deployments belonging to obj: %v", err)
+			}
+			return true
+		},
+		UpdateFunc: func(evt event.UpdateEvent) bool {
+			obj, ok := evt.ObjectOld.(*appsv1alpha1.YurtAppOverrider)
+			if !ok {
+				return ok
+			}
+			if err := r.(*ReconcileYurtAppOverrider).updatePools(obj); err != nil {
+				klog.Errorf("fail to update deployments belonging to obj: %v", err)
+			}
+			return true
+		},
+		GenericFunc: func(evt event.GenericEvent) bool {
+			return false
+		},
+	}
+
+	// Watch for changes to YurtAppOverrider
+	err = c.Watch(&source.Kind{Type: &appsv1alpha1.YurtAppOverrider{}}, &handler.EnqueueRequestForObject{}, yurtappoverriderPredicates)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// +kubebuilder:rbac:groups=apps.openyurt.io,resources=yurtappoverriders,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps,resources=deployments/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=core,resources=events,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=apps,resources=controllerrevisions,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile reads that state of the cluster for a YurtAppOverrider object and makes changes based on the state read
+// and what is in the YurtAppOverrider.Spec
+func (r *ReconcileYurtAppOverrider) Reconcile(_ context.Context, request reconcile.Request) (reconcile.Result, error) {
+
+	// Note !!!!!!!!!!
+	// We strongly recommend use Format() to  encapsulation because Format() can print logs by module
+	// @kadisi
+	klog.Infof(Format("Reconcile YurtAppOverrider %s/%s", request.Namespace, request.Name))
+
+	// Fetch the YurtAppOverrider instance
+	instance := &appsv1alpha1.YurtAppOverrider{}
+	err := r.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	if instance.DeletionTimestamp != nil {
+		return reconcile.Result{}, nil
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileYurtAppOverrider) updatePools(yao *appsv1alpha1.YurtAppOverrider) error {
```

Comment:
please update deployments that owner reference is YurtAppOverrider specified by request, instead of updating nodepools that maybe cause all deployments will be updated.